src/opm/input/eclipse/Schedule/Well/WDFAC.cpp

```diff
@@ -79,26 +82,43 @@ namespace Opm {
         if (non_trivial_dfactor) {
             m_type = WDFACTYPE::CON_DFACTOR;
         }
-    }
 
+        m_total_cf = std::accumulate(connections.begin(), connections.end(), 0.0,
+                [](const double tot_cf, const auto& conn) { return tot_cf + conn.CF(); });
+    }
 
-    double WDFAC::getDFactor(double rho, double mu, double k, double phi, double rw, double h) const {
+    double WDFAC::getDFactor(const Connection& connection, double mu, double rho, double phi, double trans_mult) const {
 
+        if (m_total_cf < 0.0) {
+            throw std::invalid_argument { "Total connection factor is not set" };
+        }
         switch (m_type)
         {
         case WDFACTYPE::NONE:
             return 0.0;
         case WDFACTYPE::DFACTOR:
-            return m_d;
+            return m_d * m_total_cf / connection.CF();
+        case WDFACTYPE::CON_DFACTOR: {
+            double d =  connection.dFactor();
+            // If a negative d factor is set in COMPDAT individual connection d factors should be used directly.
+            if (d < 0)
+                return -d;
+            // If a positive d factor is set in COMPDAT the connection d factors is treated like a well d factor.
+            // and thus scaled with the connection index
+            return d * m_total_cf / connection.CF();
+        }
         case WDFACTYPE::DAKEMODEL:
         {   
-            const auto k_md = unit::convert::to(k, prefix::milli*unit::darcy);
+            double Kh = connection.Kh() * trans_mult;
+            double Ke = connection.Ke() * trans_mult;
+            double h = Kh / Ke;
+            double rw = connection.rw();
+
+            const auto k_md = unit::convert::to(Ke, prefix::milli*unit::darcy);
             double beta = m_a * (std::pow(k_md, m_b) * std::pow(phi, m_c));
```

Comment:
This will calculate $(m\cdot K / K_0)^B = m^B \cdot (K/K_0)^B$ with `m` being the transmissibility multiplier and $K_0$ being the reference/background permeability value.  Is that intentional?

Reply:
What is K in your formula? My intention was to compute $(m\cdot Ke )^B$ where Ke = connection.Ke() and thus include the multiplier in the dake formula. Do you want me to remove the multiplier? I am not sure about the inclusion of the multiplier in the computation of beta, but I think it should be in the last return i.e. `beta * specific_gravity * Ke / (h * mu * rw );`

Reply:
> What is K in your formula?

The effective permeability.

> My intention was to compute $(m\cdot K_e)^B$ and thus include the multiplier in the Dake formula. 

Okay, understood.  I believe that's wrong, and to be honest I don't think the transmissibility multiplier should go into this formula at all.  My impression is that apart from the viscosity, there is no dynamic effects in the Dake formula.

Reply:
Ok. I see your point. I will remove it.